### ISSUE TITLE
what is happening if the loss of generator keeps increasing?

### ISSUE THREAD

**Issue by ypxie** [AlexiaJM/Deep-learning-with-cats#10]
for the wgan-gp?
Thanks!

**Comment by AlexiaJM** [comment]
We penalize the discriminator so the loss of G shouldn't increase too much. Do you use the default settings? Btw, to run WGAN-GP you need the bleeding-edge version of Pytorch because backward of gradients is not yet in the stable release.

**Comment by ypxie** [comment]
Thanks for your reply.  I updated my pytorch.
By the way. Could you also elaborate more about why did the current code do backward two times (one for real,  and one for fake) rather than do total_loss.backward() once? Are there any specific reasons?

**Comment by AlexiaJM** [comment]
Hi ypxie,

This is because the loss function include D, G and the penalty of the gradient. Once we got everything added to the gradients (from the backward steps), we minimize the loss. See formula (3) on p4:
https://arxiv.org/pdf/1704.00028.pdf

**Comment by ypxie** [comment]
Thanks for your reply.
I mean why don't you directly use the following? Isn't this more efficient?
`
  d_loss = err_fake - err_real +  grad_penalty
  d_loss.backward() `

**Comment by AlexiaJM** [comment]
In big O complexity, it would be the same. If you think that in practice, it would be faster, you can try and benchmark it and if you show that it's faster, make a pull request and we'll change it.

**Comment by ypxie** [comment]
Thank you for your reply. 
Did you also try the whole meow dataset besides the cat face?

**Comment by AlexiaJM** [comment]
That's what I did the first time I tried but it wouldn't converge. This was using DCGAN though and using old code. You can always retry but this is likely not to converge because it's a too complex and noisy thing to model, especially with a small dataset (10k is small for pictures). So far, GANs trying to generate images with imagenet tends to create images that don't make much sense or are extremely low resolution. We're are still not there yet (someone correct me if I wrong).

**Comment by ypxie** [comment]
Thanks for your reply. I also find it very difficult to get meaningful images using the full meow dataset.  I am now trying some more complex models, I will let you know if I could generate nice images.
